src/ralph/reports/views.py

```diff
@@ -453,3 +454,36 @@ def prepare(self, model, *args, **kwargs):
                 yield base_row + fill_empty_assets + row + [
                     single_licence_cost
                 ]
+
+
+class FailureReport(ReportWithoutAllModeDetail, ReportDetail):
+    with_datacenters = True
+    name = _('Failures')
+    description = _('Failure types for each manufacturer.')
+
+    def prepare(self, model, dc=None):
+        queryset = model._default_manager
+        if dc:
+            queryset = queryset.filter(rack__server_room__data_center=dc)
+        operation_types = OperationType.objects.get(
+            pk=OperationType.choices.failure
+        ).get_descendants(include_self=True)
+        failures = Failure.base_objects.through.objects.filter(
+            baseobject__in=queryset.all(),
+            operation__type__in=operation_types
```

Comment:
it's complicated. Actually, we don't use `Failure`'s manager here - it's not `Failure`'s objects, but `base_objects.through` objects, which is M2M relationship intermediate table and we store there every `Operation` relation.